ga4gh/backend.py

```diff
@@ -214,113 +299,44 @@ def variantSetsGenerator(self, request):
             request, self._variantSetIdMap, self._variantSetIds)
 
     def readsGenerator(self, request):
-        # Local utility functions to save some typing
-        def getPosition(readAlignment):
-            return readAlignment.alignment.position.position
-
-        def getEndPosition(readAlignment):
-            return getPosition(readAlignment) + \
-                len(readAlignment.alignedSequence)
-
-        if len(request.readGroupIds) != 1:
-            raise exceptions.NotImplementedException(
-                "Read search over multiple readGroups not supported")
-        readGroupId = request.readGroupIds[0]
-        try:
-            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
-        except KeyError:
-            raise exceptions.ReadGroupNotFoundException(readGroupId)
-        startPosition = request.start
-        equalPositionsToSkip = 0
-        if request.pageToken is not None:
-            startPosition, equalPositionsToSkip = self.parsePageToken(
-                request.pageToken, 2)
+        """
+        Returns a generator over the (read, nextPageToken) pairs defined
+        by the specified request
+        """
+        readGroup = self._getReadGroup(request)
+        startPosition, equalPositionsToSkip = _getIntervalCounters(request)
```

Comment:
Ah, OK, I've just seen the reason why we need `startPosition`; sorry about that.

Agreed on the subclassing idea, that is a much nicer design.